doc/how_to/custom_components/esm/build.md

```diff
@@ -0,0 +1,161 @@
+
+    _importmap = {
+        "imports": {
+            "canvas-confetti": "https://esm.sh/canvas-confetti@1.6.0",
+        }
+    }
+
+    _esm = 'confetti.js'
+```
+
+Now that we have separately declared the import we can update the `import` line in the `confetti.js` file:
+
+```javascript
+/* confetti.js */
+import confetti from "canvas-confetti";
+```
+
+This approach cleanly separates the definitions of the libraries and their versions from the actual code. Import maps have a bunch of other features but in most cases the imports section will be all you need.
+
+## Bundling
+
+Importing libraries directly from a CDN allows for extremely quick iteration but also means that the users of your components will have to have access to the internet to fetch the required modules. By bundling the component resources you can ship a self-contained module that includes all the dependencies, while also ensuring that you only fetch the parts of the libraries that are actually needed.
+
+### Tooling
+
+The tooling we recommend to bundle your component resources include `esbuild` and `npm`, both can conveniently be installed with `conda`:
+
+```bash
+conda install esbuild npm
+```
+
+### Configuration
+
+To run the bundling we will need one additional file, the `package.json`, which, just like the import maps, determines the required packages and their versions. The `package.json` is a complex file with tons of configuration options but all we will need are the [dependencies](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#dependencies).
+
+To recap here are the three files that we need:
+
+::::{tab-set}
+
+:::tab-item package.json
+```json
+{
+  "name": "confetti-button",
+  "dependencies": {
+    "canvas-confetti": "^1.6.0"
+  }
+}
+```
+:::
+
+:::{tab-item} confetti.py
+import panel as pn
+
+from panel.custom import JSComponent
+
+pn.extension()
+
+class ConfettiButton(JSComponent):
+
+    _esm = 'confetti.bundled.js'
+
+ConfettiButton().servable()
+:::
+
+:::{tab-item} confetti.js
+import confetti from "canvas-confetti";
+
+export function render() {
+  const button = document.createElement('button')
+  button.addEventListener('click', () => confetti())
+  button.append('Click me!')
+  return button
+}
+:::
+
+::::
+
+Once you have set up these three files you have to install the packages with `npm`:
+
+```bash
+npm install
+```
+
+This will fetch the packages and install them into the local `node_modules` directory. Once that is complete we can run the bundling:
+
+```bash
+esbuild confetti.js --bundle --format=esm --minify --outfile=confetti.bundled.js
+```
+
+This will create a new file called `confetti.bundled.js`, which includes all the dependencies (even CSS, image files and other static assets if you have imported them).
+
+The only thing left to do now is to update the `_esm` declaration to point to the new bundled file:
+
+```python
+class ConfettiButton(JSComponent):
+
+    _esm = 'confetti.bundled.js'
+```
```

Comment:
Would be nice to one day have a reference repository for creating and distribution a custom component like this as a package.